application/common/models/User.php

```diff
@@ -594,45 +595,61 @@ public function getNagState()
                 $this->nag_for_mfa_after,
                 $this->nag_for_method_after,
                 $this->review_profile_after,
-                count($this->getVerifiedMfaOptions()),
+                $this->getVerifiedMfaOptionsCount(),
                 count($this->getVerifiedMethodOptions())
             );
         }
 
         return $this->nagState->getState();
     }
 
-
-    /**
-     * @return array MFA related properties
-     */
-    public function getMfaFields()
+    public function loadMfaData(string $rpOrigin = '')
     {
-        return [
+        $verifiedMfaOptions = $this->getVerifiedMfaOptions($rpOrigin);
+        $this->mfa = [
             'prompt'  => $this->isPromptForMfa() ? 'yes' : 'no',
             'add'     => $this->getNagState() == NagState::NAG_ADD_MFA ? 'yes' : 'no',
-            'active'  => count($this->getVerifiedMfaOptions()) > 0 ? 'yes' : 'no',
-            'options' => $this->getVerifiedMfaOptions(),
+            'active'  => count($verifiedMfaOptions) > 0 ? 'yes' : 'no',
+            'options' => $verifiedMfaOptions,
         ];
     }
 
     /**
+     * WARNING: Every call to this DURING authentication will trigger one or
+     * more calls to our MFA API, to initialize an authentication for each
+     * verified MFA option. If any of those MFAs are WebAuthn, the RP Origin
+     * must be provided for the call to succeed.
+     *
+     * If all you need is a count, use `getVerifiedMfaOptionsCount()` instead.
+     *
      * @return Mfa[]
      */
-    public function getVerifiedMfaOptions()
+    public function getVerifiedMfaOptions(string $rpOrigin = ''): array
     {
         $mfas = [];
         foreach ($this->mfas as $mfaOption) {
             if ($mfaOption->verified === 1) {
                 if ($this->scenario == self::SCENARIO_AUTHENTICATE || $mfaOption->type !== Mfa::TYPE_MANAGER) {
                     $mfaOption->scenario = $this->scenario;
+                    $mfaOption->loadData($rpOrigin);
                     $mfas[] = $mfaOption;
                 }
             }
         }
         return $mfas;
     }
 
+    public function getVerifiedMfaOptionsCount(): int
+    {
+        $count = 0;
+        foreach ($this->mfas as $mfaOption) {
+            if ($mfaOption->verified === 1) {
```

Comment:
Don't change this for the release (low importance) ...  I'm wondering if this doesn't need the `if` statement, just
 `$count += $mfaOption->verified`